src/main/java/me/itzg/tsdbcassandra/services/DownsampleProcessor.java

```diff
@@ -80,28 +84,55 @@ public void setupSchedulers() {
       throw new IllegalStateException("Downsample partitions to process are configured, but not granularities");
     }
 
-    scheduled = Flux.interval(downsampleProperties.getDownsampleProcessPeriod(), scheduler)
-        .flatMap(tick -> Flux.fromIterable(downsampleProperties.getPartitionsToProcess()))
-        .flatMap(this::processPartition)
-        .subscribe();
+    scheduled = downsampleProperties.getPartitionsToProcess().stream()
+        .map(partition ->
+            taskScheduler.scheduleAtFixedRate(
```

Comment:
Before I added the `LIMIT` on the pending downsample set retrieval, the downsample results insertion was overwhelming the single node Cassandra running locally. In the process I discovered that Reactor error handling was a little more aggressive than I expected in causing the whole publisher stream to error out. As a result, the `Flux.interval` was getting cancelled the moment there was any error. I tried some error recovery operators without success, so I gave up and used plain old task scheduling here.